mmv1/third_party/terraform/resources/resource_cloudfunctions_function.go

```diff
@@ -146,6 +146,18 @@ func resourceCloudFunctionsFunction() *schema.Resource {
 				},
 			},
 
+			"docker_repository": {
```

Comment:
Is this in GA or Beta? It looks like dockerRepository and kmsKeyName are both [in the v1 GA API](https://cloud.google.com/functions/docs/reference/rest/v1/projects.locations.functions), which is what this resource calls - but the tests are currently gated behind `<% unless version == "ga" %>` and the docs call these fields out as GA. If the fields are actually beta they should be gated as well; otherwise the other places should be revised to match GA support.

Reply:
It's GA for GCF CMEK, although Terraform support for one of its dependencies, Artifact Registry, is still in beta (the AR itself is in the GA). I had the same question while implementing this, checked with Terraform team -- while this is unusual, it sounded like this may be acceptable and there turned out to be precedents for such cases. For instance, "workload identity pool is beta-only, but can be used for a GA feature of IAM as seen here": https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/tests/resource_google_service_account_iam_test.go.erb#L230

As for tests, those for "GA field that depends on a beta feature won't be able to be run in the GA provider, but they can be tested in the beta provider". So, I have intentionally removed "unless GA" gating on the actual changes.

Reply:
If this can be used in GA without artifact repository, could you add a test for GA (and Beta) that will exercise these fields?

Reply:
The GCF CMEK doesn't work without Artifact Registry, because the other registry type (GCR) does not support CMEK.  Basically, there is no way to use it without referring to the AR, which is still in the beta provider and that didn't work well in the GA repo. Once the AR field advances to the GA, we can remove the guarding from the beta test to extend the coverage.

Reply:
checked on this and it's okay for the field to be GA even though it relies on a beta feature.